src/ensembl/src/shared/components/external-reference/ExternalReference.test.tsx

```diff
@@ -37,45 +37,53 @@ describe('<ExternalReference />', () => {
     props: Partial<ExternalReferenceProps> = {}
   ) => render(<ExternalReference {...defaultProps} {...props} />);
 
-  let container: any;
-
   beforeEach(() => {
     jest.resetAllMocks();
-    container = renderExternalReference().container;
   });
 
   it('renders without error', () => {
+    const { container } = renderExternalReference();
     expect(() => container).not.toThrow();
   });
 
   it('hides label container div when there is no label', () => {
-    container = renderExternalReference({ label: undefined }).container;
-    expect(container.querySelectorAll('.label')).toHaveLength(0);
+    const { container } = renderExternalReference({ label: undefined });
+
+    expect(container.querySelector('.label')).toBeFalsy();
   });
 
   it('applies the passed in classNames', () => {
+    const { container } = renderExternalReference();
+
     expect(
-      screen
-        .getByTestId('external reference container')
-        .classList.contains(defaultProps.classNames?.container as string)
+      container.querySelector(`.${defaultProps.classNames?.container}`)
     ).toBeTruthy();
 
     expect(
       container
         .querySelector('.label')
-        .classList.contains(defaultProps.classNames?.label)
+        ?.classList.contains(defaultProps.classNames?.label as string)
     ).toBeTruthy();
 
-    const externalLink = screen.getByTestId('external link container');
     expect(
-      externalLink
-        .getElementsByTagName('icon-mock')[0]
-        .getAttribute('classname')
-    ).toMatch(defaultProps.classNames?.icon as string);
+      container
+        .querySelector(`.externalLinkcontainer icon-mock`)
+        ?.getAttribute('classname')
+        ?.indexOf(defaultProps.classNames?.icon as string)
+    ).toBeTruthy();
+
     expect(
-      externalLink
-        .getElementsByTagName('a')[0]
-        .classList.contains(defaultProps.classNames?.link as string)
+      container.querySelector(
+        `.externalLinkcontainer .${defaultProps.classNames?.link}`
+      )
     ).toBeTruthy();
   });
+
+  it('does not display ExternalLink component when there is no link', () => {
+    const { container } = renderExternalReference({ to: undefined });
+
+    expect(container.querySelector(`.externalLinkcontainer`)).toBeFalsy();
+
+    expect(container.querySelector(`.noLink`)).toBeTruthy();
```

Comment:
probably nice to make sure that it contains the linkText as well? 😉

Reply:
That's done